### ISSUE TITLE
Desired images from survey

### ISSUE THREAD

**Issue by scollis** [ARM-DOE/pyart-roadmap#15]
Please use this tracker to share the plots you have made so far

I need the following:
two bar chart with respondents who have hand have not used Py-ART

For both groups a breakdown of where they come from (academia, labs etc.. ) And also the "roll" ie student, postdoc et al..

For both of these combine the non-pyart and pyart groups. We will NOT be trying to break them into model/obs users.

**Comment by zssherman** [comment]
Hi Sc0tt,

C0ry and I are working on the charts and wanted your opinion. We can always change the color, but these are the usage results.

![usage-barchart](https://cloud.githubusercontent.com/assets/19153455/18685860/6dfdddd8-7f3f-11e6-85a1-280e95739bd6.png)
![usage-piechart](https://cloud.githubusercontent.com/assets/19153455/18685861/6e096d2e-7f3f-11e6-9fa1-6ac9b14b6682.png)

**Comment by scollis** [comment]
top

Dr Scott Collis
Purveyor of lofty observations (Atmospheric Scientist) 
Environmental Science Division
Argonne National Laboratory
Mb: +1 630 235 8025
Of: +1 630 252 0550
Become a Py-ART user today! http://arm-doe.github.io/pyart/ http://arm-doe.github.io/pyart/

**Comment by zssherman** [comment]
Sorry late response we were trying to figure out a couple of the other plots,

top it is, any color preference?

**Comment by scollis** [comment]
Bw is good

Can you aggregate statistics for both pyart and non pyart users for affiliation (e.g. University etc) and position (student et al) 
Then we will present this as a three panel plot 

-sent from a mobile device-

**Comment by zssherman** [comment]
Yeah that sounds good, we can do that.

**Comment by Rumpkie** [comment]
We made really good progress on combining the data and recalculating the ratings. tomorrow we should have a number of very nice looking graphs for you.
-C

**Comment by scollis** [comment]
Great work 

-sent from a mobile device-

**Comment by zssherman** [comment]
Hi Scott, 
Still trying to figure out how to add space between the bars, but how is this for new features? Or would you want two seperate plots?

![nonuser_user_favoritefeatures_barchart](https://cloud.githubusercontent.com/assets/19153455/18723207/4bab4ef0-7ffd-11e6-85d8-4aab0fd1ffff.png)

**Comment by scollis** [comment]
Create two separate images.. we can merge them in post.

Play with figsize in plt.figure(figsize=[x,y])

I would also suggest adding a ‘\n’ in some of the categories or abbreviate them… like “Add Cartopy…” 

@rumpkie is busy today but will meet soon to sort out the inverse ranking.. Like the way the layout makes clear how much more some things are popular.

Dr Scott Collis
Purveyor of lofty observations (Atmospheric Scientist) 
Environmental Science Division
Argonne National Laboratory
Mb: +1 630 235 8025
Of: +1 630 252 0550
Become a Py-ART user today! http://arm-doe.github.io/pyart/ http://arm-doe.github.io/pyart/

**Comment by zssherman** [comment]
Ok that sounds good.

**Comment by zssherman** [comment]
More plots,

![combined_organization_belong_piechart](https://cloud.githubusercontent.com/assets/19153455/18754903/dbcd9ba2-80af-11e6-9924-140b98b07e48.png)
![combined_organization_role_piechart](https://cloud.githubusercontent.com/assets/19153455/18754909/dd5dc2f8-80af-11e6-8fc7-32fc4192e21b.png)

**Comment by scollis** [comment]
perfect!

Dr Scott Collis
Purveyor of lofty observations (Atmospheric Scientist) 
Environmental Science Division
Argonne National Laboratory
Mb: +1 630 235 8025
Of: +1 630 252 0550
Become a Py-ART user today! http://arm-doe.github.io/pyart/ http://arm-doe.github.io/pyart/

**Comment by zssherman** [comment]
Fixed bar plots and user favorite feature plot,

![non-user_new_feature_barchart](https://cloud.githubusercontent.com/assets/19153455/18793974/ee5b5f74-8182-11e6-8dfa-b74186fb8e6c.png)

![user_favorite_feature_barchart](https://cloud.githubusercontent.com/assets/19153455/18793981/f51a6242-8182-11e6-9785-abef483b7807.png)

![user_new_feature_barchart](https://cloud.githubusercontent.com/assets/19153455/18793985/f8eb8982-8182-11e6-9b64-c4902b3d00f0.png)

**Comment by scollis** [comment]
VERY happy

Ok.. one change: Get rid of the “figure…” bit at the bottom.. I will do that in the text

Dr Scott Collis
Purveyor of lofty observations (Atmospheric Scientist) 
Environmental Science Division
Argonne National Laboratory
Mb: +1 630 235 8025
Of: +1 630 252 0550
Become a Py-ART user today! http://arm-doe.github.io/pyart/ http://arm-doe.github.io/pyart/

**Comment by zssherman** [comment]
Ok sounds good, I'll remove the figure text and upload  the plots to git.

**Comment by scollis** [comment]
BTW.. **Don't** add these to /roadmap/images  I will add them in my PR

**Comment by zssherman** [comment]
Gotcha, after removing the figure text should I just re add them in this thread?

**Comment by scollis** [comment]
Yes.. Probably not the intended use of GitHub issues but this is working very well

Dr Scott Collis
Purveyor of lofty observations (Atmospheric Scientist) 
Environmental Science Division
Argonne National Laboratory
Mb: +1 630 235 8025
Of: +1 630 252 0550
Become a Py-ART user today! http://arm-doe.github.io/pyart/ http://arm-doe.github.io/pyart/

**Comment by zssherman** [comment]
![usage-barchart](https://cloud.githubusercontent.com/assets/19153455/18796520/4f0e03ec-818f-11e6-8b40-98fc57e249fe.png)

![usage-piechart](https://cloud.githubusercontent.com/assets/19153455/18796525/52cd2a3a-818f-11e6-8e78-9f6c63b07929.png)

![non-user_new_feature_barchart](https://cloud.githubusercontent.com/assets/19153455/18796535/5d2740ba-818f-11e6-8a09-410db9e1cf89.png)

![user_new_feature_barchart](https://cloud.githubusercontent.com/assets/19153455/18796548/690db13e-818f-11e6-9d12-ce14a771b2dc.png)

![user_favorite_feature_barchart](https://cloud.githubusercontent.com/assets/19153455/18796538/617627da-818f-11e6-9ebc-06ff0ba989e7.png)

![combined_organization_belong_piechart](https://cloud.githubusercontent.com/assets/19153455/18796551/7003f3cc-818f-11e6-9e94-17aced9cc50f.png)

![combined_organization_role_piechart](https://cloud.githubusercontent.com/assets/19153455/18796555/727065c8-818f-11e6-9125-1be7a809b721.png)

**Comment by zssherman** [comment]
Forgot to change the title,

![usage-piechart](https://cloud.githubusercontent.com/assets/19153455/18796608/aaf2fadc-818f-11e6-8c06-c439dc2394a3.png)

**Comment by scollis** [comment]
Very nice..

**Comment by zssherman** [comment]
Yay!

**Comment by Rumpkie** [comment]
These have come out great @zssherman, well done.

**Comment by scollis** [comment]
Ok.. So now your task @zssherman is to see if you can pull the data out of the GitStats module. So I think making it log scale may fix it I am up to suggestions but I love the idea of a plot that
- Shows the growth of the project over time
- highlights user contributions..

This may take some hacking.. But see if you could:
- Plot only Python language contributions
- Plot number of commits by user

Basically as mentioned before I want to show not only the growth of the code base but those contributing before it.. @jjhelmus and I will be in Oak Ridge next week. But please use @Rumpkie for advice.

**Comment by zssherman** [comment]
Ok that sounds good.

**Comment by scollis** [comment]
Hey @zssherman I also need a chart for "barriers to contributing" 
you can do this Monday.. @Rumpkie knows what i am talking about

**Comment by zssherman** [comment]
Can do,
and did you want the non-user barriers to using Py-ART also?

**Comment by scollis** [comment]
As a low priority.. 

Priority:
1) Py-ART users barrier to contrib
2) Better history plot
3) Non-Py-ART barriers to entry

Dr Scott Collis
Purveyor of lofty observations (Atmospheric Scientist) 
Environmental Science Division
Argonne National Laboratory
Mb: +1 630 235 8025
Of: +1 630 252 0550
Become a Py-ART user today! http://arm-doe.github.io/pyart/ http://arm-doe.github.io/pyart/

**Comment by zssherman** [comment]
Gotcha, ok sounds good.

**Comment by zssherman** [comment]
![user_barriers_contribute_barchart](https://cloud.githubusercontent.com/assets/19153455/18837698/5c05c458-83ca-11e6-9087-d0017e77396f.png)

**Comment by zssherman** [comment]
Added barriers to contribute plot and made fonts larger and removed some space between the bars.

![non-user_new_feature_barchart](https://cloud.githubusercontent.com/assets/19153455/18840010/ccc774a4-83d2-11e6-9dd9-4b0ff5388d9f.png)

![user_new_feature_barchart](https://cloud.githubusercontent.com/assets/19153455/18840023/d747e24c-83d2-11e6-91e0-28c6870b3bce.png)

![user_favorite_feature_barchart](https://cloud.githubusercontent.com/assets/19153455/18840042/e80e6240-83d2-11e6-9463-cee2c5d7e0b8.png)

![user_barriers_contribute_barchart](https://cloud.githubusercontent.com/assets/19153455/18840058/f40b6318-83d2-11e6-8a96-f5fb476e622e.png)

**Comment by zssherman** [comment]
I was able to get at the data from the gitstats results, I'm still working on how to graph just python contributions, but I changed the current graphs to log scales.

![lines_authors_overtime](https://cloud.githubusercontent.com/assets/19153455/18850116/b277dd54-83fb-11e6-97bc-ebb1784a3446.png)

![commits_authors_overtime](https://cloud.githubusercontent.com/assets/19153455/18850121/b5b79ad6-83fb-11e6-8866-ee25a41e8f7d.png)

![lines_overtime](https://cloud.githubusercontent.com/assets/19153455/18850151/d58ba1ea-83fb-11e6-973d-b7ed4b4264f9.png)

**Comment by scollis** [comment]
Fantastic!

-sent from a mobile device-

**Comment by zssherman** [comment]
![nonuser_barriers_using_pyart_barchart](https://cloud.githubusercontent.com/assets/19153455/18851555/2f4560e4-8402-11e6-943a-f079b5781f1c.png)

**Comment by zssherman** [comment]
Increased line width in the legend, changed the colors making it somewhat easier to distiguish between each, and added a grey background to help the brighter colors to stand out more.

![commits_authors_overtime](https://cloud.githubusercontent.com/assets/19153455/18878143/2e7e950c-8495-11e6-8e66-66265668f43a.png)

![lines_authors_overtime](https://cloud.githubusercontent.com/assets/19153455/18878144/306ad4c0-8495-11e6-948e-ec14b9d399d7.png)

**Comment by scollis** [comment]
Very nice!

Dr Scott Collis
Purveyor of lofty observations (Atmospheric Scientist) 
Environmental Science Division
Argonne National Laboratory
Mb: +1 630 235 8025
Of: +1 630 252 0550
Become a Py-ART user today! http://arm-doe.github.io/pyart/ http://arm-doe.github.io/pyart/

**Comment by scollis** [comment]
Updated request:

Two panel plot with one common legend.. make it wide format (i.e. to span across the whole page). 

Dr Scott Collis
Purveyor of lofty observations (Atmospheric Scientist) 
Environmental Science Division
Argonne National Laboratory
Mb: +1 630 235 8025
Of: +1 630 252 0550
Become a Py-ART user today! http://arm-doe.github.io/pyart/ http://arm-doe.github.io/pyart/

**Comment by zssherman** [comment]
Ok sounds good, and just in case here are the two individual images but larger and match the legend size. But i'll start working on the two panel plot. 

![commits_authors_overtime](https://cloud.githubusercontent.com/assets/19153455/18880046/2f269e80-849c-11e6-8ebb-eb8e269057e2.png)

![lines_authors_overtime](https://cloud.githubusercontent.com/assets/19153455/18880059/32da49a0-849c-11e6-9e32-ed41b2da417d.png)

**Comment by zssherman** [comment]
Such as this?

![combined_lines_commits_overtime](https://cloud.githubusercontent.com/assets/19153455/18881439/e11d5904-84a0-11e6-8981-773f6bc5cefd.png)

**Comment by scollis** [comment]
Yes :) That!

Dr Scott Collis
Purveyor of lofty observations (Atmospheric Scientist) 
Environmental Science Division
Argonne National Laboratory
Mb: +1 630 235 8025
Of: +1 630 252 0550
Become a Py-ART user today! http://arm-doe.github.io/pyart/ http://arm-doe.github.io/pyart/

**Comment by zssherman** [comment]
That image i'll email to you, it seems when I add it here it is hard to read.
